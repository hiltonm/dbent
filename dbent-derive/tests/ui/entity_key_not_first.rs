use dbent::prelude::*;

#[derive(Entity)]
struct Test {
    data: String,
    id: Key<Int>,
}

fn main() {}
