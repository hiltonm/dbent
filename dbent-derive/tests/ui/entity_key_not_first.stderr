error: #[derive(Entity)] needs the first field to be a Key; aliasing the Key to something else breaks the macro
 --> tests/ui/entity_key_not_first.rs:5:5
  |
5 |     data: String,
  |     ^^^^
